### PR TITLE
Feature/revamp bootstrapping

### PR DESCRIPTION
**Pull request by amrox**
This allows zinc to read files and serve files directly from the main app bundle, without copying them into the repo first.

It will make SHA-based symlink to the files in the main app bundle, so Zinc will know it doesn't need to download them.

### CODE REVIEW

File: Zinc/ZincRepoIndex.m [hunk 1]
```diff
@@ -201,6 +209,27 @@ - (void) removeBundle:(NSURL*)bundleResource
     }
 }
 
+- (void) registerExternalBundle:(NSURL*)bundleRes rootPath:(NSString*)rootPath
+{
+    @synchronized(self.myExternalBundleRefs) {
+        self.myExternalBundleRefs[bundleRes] = rootPath;
```
Comment (JaviSoto) [inline Zinc/ZincRepoIndex.m:215]: Remember this syntax is only supported since Xcode 4.5 :)

**Comment by JaviSoto** [comment]
Looks good.

**Comment by amrox** [comment]
I wish it had more tests :/ But it will have to do

**Comment by kstenerud** [comment]
:shipit:

**Pull request merged**